lib_test/unit_tests.ml

```diff
@@ -264,6 +262,53 @@ let test_ca_325844 () =
   let rrd = Rrd_unix.of_file (Filename.concat "test_data"  "flip_flop.xml") in
   test_ranges rrd ()
 
+let suite_create_multi =
+  let module RU = Rrd_updates in
+
+  let assert_size t =
+    (** we can't to check that the number of rows is consistent,
+        since this is defined purely by the number of rows
+      * the number of columns should match the number of items in the legend -
+        each element in the legend array defines the contents for one column *)
+    let num_cols_in_legend = Array.length t.RU.legend in
+    t.RU.data |> Array.iteri (fun i r ->
+      Alcotest.check Alcotest.int
+        (Printf.sprintf "number of cols in legend matches number of cols in row#%i" i)
+        num_cols_in_legend
+        (Array.length r.RU.row_data)
+    )
+  in
+  let test_no_rrds () =
+    Alcotest.check_raises "should raise error" (Failure "hd") (fun () -> let _ = RU.create_multi [] 0L 1L None in ())
+  in
+  let test_one_rrd () =
+    let one_rrd = create_rrd [0L; 5L; 10L;] 0. 1. in
+    RU.create_multi ["one_rrd", one_rrd] 0L 1L None |> assert_size
+  in
+  (** confusingly, rows in an rra are used to define the cols in the rrd_updates/ xml...
+    * essentially we usually expect 'rows' in each rrd to be the same (test_rows_with_same_num_cols)
+    * however, we should also handle the case where they are not (test_rows_with_different_num_cols) *)
+  let test_rows_with_same_num_cols () =
+    RU.create_multi
+      [ "row", create_rrd ~rows:3 [0L; 5L; 10L] 0. 1.
+      ; "another_row", create_rrd ~rows:3 [1L; 6L; 11L] 0. 1.
+      ] 0L 1L None
+    |> assert_size
+  in
+
+  let test_rows_with_different_num_cols () =
+    RU.create_multi
+      [ "row", create_rrd ~rows:2 [0L; 5L; 10L] 0. 1.
+      ; "dodgy_row", create_rrd ~rows:3 [1L; 6L; 11L] 0. 1.
+      ] 0L 1L None
+    |> assert_size
```

Comment:
Instead of this I recommend to define the data separately from the checks, something like
```
let updates_with_valid_size = [
  "one rrd", RU.create_multi ["one_rrd"] ....
; "rows_with_same_num_cols", RU.create_multi ...
; "rows_with_different_num_cols", RU.create_multi
    (* confusingly, rows ...*)  
    [ "row" ...
    [ "different_length_row" ...
]

let test_valid_length_updates multi () = assert_size multi

List.map (fun name, multi ->
  name, `Quick, test_valid_length_updates multi
  ) updates_with_valid_size
```
See some example in https://github.com/xapi-project/xen-api/blob/feature/REQ-453/master/ocaml/tests/test_certificates.ml#L24-L27

Reply:
that is neater :)